### PR TITLE
Add CMake support

### PR DESCRIPTION
**Pull request by sloretz**

### CODE REVIEW

File: drake_ros/CMakeLists.txt [hunk 1]
```diff
@@ -19,10 +19,17 @@ find_package(pybind11 REQUIRED HINTS "${drake_DIR}/../pybind11" NO_DEFAULT_PATH)
 add_subdirectory(core)
 add_subdirectory(tf2)
 add_subdirectory(viz)
-# Python bindings.
+# Python bindings
 add_subdirectory(drake_ros)
 
+# Install python package must happen at top level CMakeLists.txt
```
Comment (EricCousineau-TRI) [inline drake_ros/CMakeLists.txt:25]: Not sure if I understand this part - why must it happen at top-level, but included subdirectory has an install directive for Python stuff?

File: drake_ros/CMakeLists.txt [hunk 2]
```diff
@@ -19,10 +19,17 @@ find_package(pybind11 REQUIRED HINTS "${drake_DIR}/../pybind11" NO_DEFAULT_PATH)
...
+# TODO(sloretz) exclude drake_ros/test folder
 ament_python_install_package(drake_ros)
+# Generate same python package in build directory for use in tests
+file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/py/")
```
Comment (EricCousineau-TRI) [inline drake_ros/CMakeLists.txt:29]: This smells... weird? Is there a bug underlying this? Or is it a common convention for ROS 2 packages?

**Review by EricCousineau-TRI — CHANGES_REQUESTED** [review]
Sweet, thanks! Two q's

File: drake_ros/CMakeLists.txt (hunk 2 above)
Comment (sloretz) [reply drake_ros/CMakeLists.txt:29]: With CMake there are three directories. There's the place where source code is (`*_SOURCE_DIR`), there's the place where build artifacts go (`*_BINARY_DIR`), and there's a folder that artifacts get installed into (`CMAKE_INSTALL_PREFIX/*`). When we're testing code we have access to two of them: source and binary. When we're testing the python code we want to be able to import both the Python code and the cPython extentions. However, the python code is in the source directory, while the cpython extension is in the binary directory.

This copies the pure python files into the binary directory. Elsewhere the target property `LIBRARY_OUTPUT_DIRECTORY` is set so the cPython extension is put into the right spot for Python to be able to import it. The result is tests can import everything from the binary directory.

If we only had pure Python code, we wouldn't need to make a copy as the tests could import it from the source directory.

File: drake_ros/CMakeLists.txt (hunk 1 above)
Comment (sloretz) [reply drake_ros/CMakeLists.txt:25]: The `ament_python_install_package()` call has to be in the top level `CMakeLists.txt`. It sets variables in the parent scope which are picked up by a hook in `ament_package()`. `add_subdirectory()` creates a new scope in CMake, so `ament_python_install_package()` can't be put into it as `ament_package()` wouldn't see the variables being set.

Comment (EricCousineau-TRI) [reply drake_ros/CMakeLists.txt:25]: Gotcha. In that case, can you qualify the comment, e.g.

```
# Ament installation must happen in top-level, including Python package installation
```

File: drake_ros/CMakeLists.txt (hunk 2 above)
Comment (EricCousineau-TRI) [reply drake_ros/CMakeLists.txt:29]: Hm... I guess I'm surprised that the build folder matters? I would have thought that `colcon test` would leverage everything in the install tree?

**Comment by EricCousineau-TRI** [comment]
Will go ahead and merge - I'll address the "change comment".

However, I'm still surprised about possible (broken?) redundancy between install tree and build tree for `colcon test`.

File: drake_ros/CMakeLists.txt (hunk 2 above)
Comment (EricCousineau-TRI) [reply drake_ros/CMakeLists.txt:29]: Per VC, this seems like an expected dev pattern. Unclear if this is documented

**Pull request merged**